flutter/lib/src/flutter_enricher.dart

```diff
@@ -0,0 +1,199 @@
+import 'package:sentry/sentry.dart';
+
+typedef WidgetBindingGetter = WidgetsBinding Function();
+
+/// Enriches [SentryEvent]s with various kinds of information.
+/// FlutterEnricher only needs to add information which aren't exposed by
+/// the Dart runtime.
+class FlutterEnricher implements Enricher {
+  FlutterEnricher(
+    this._checker,
+    this._dartEnricher,
+    this._getWidgetsBinding,
+  );
+
+  factory FlutterEnricher.simple(PlatformChecker checker) {
+    return FlutterEnricher(
+      checker,
+      Enricher(),
+      () => WidgetsFlutterBinding.ensureInitialized(),
+    );
+  }
+
+  // We can't use `WidgetsBinding` as a direct parameter
+  // because it must be called inside the `runZoneGuarded`-Integration.
+  // Thus we call it on demand after all the initialization happend.
+  final WidgetBindingGetter _getWidgetsBinding;
+  WidgetsBinding get _widgetsBinding => _getWidgetsBinding();
+  final PlatformChecker _checker;
+  final Enricher _dartEnricher;
+  SingletonFlutterWindow get _window => _widgetsBinding.window;
+  Map<String, String> _packages = {};
+
+  @override
+  FutureOr<SentryEvent> apply(
+    SentryEvent event,
+    bool hasNativeIntegration,
+    bool includePii,
+  ) async {
+    // Flutter for Web does not need a special case.
+    // It's already covered by _dartEnricher.
+
+    // First use _dartEnricher.
+    // In case we have even better information available in Flutter
+    // we override what's already given by _dartEnricher.
+    event = await _dartEnricher.apply(event, hasNativeIntegration, includePii);
+
+    // If there's a native integration available, it probably has better
+    // information available than Flutter.
+    final device =
+        hasNativeIntegration ? null : _getDevice(event.contexts.device);
+
+    final contexts = event.contexts.copyWith(
+      device: device,
+      runtimes: _getRuntimes(event.contexts.runtimes),
+    );
+
+    // Flutter has a lot of Accessibility Settings available and exposes them
+    contexts['accessibility'] = _getAccessibilityContext();
+
+    // Conflicts with Flutter runtime if it's just called `Flutter`
+    contexts['flutter_context'] = _getFlutterContext();
+
+    contexts['culture'] = _getCulture();
+
+    return event.copyWith(
+      contexts: contexts,
+      modules: await _getPackages(),
+    );
+  }
+
+  /// Packages are loaded from [LicenseRegistry].
+  /// This is currently the only way to know which packages are used.
+  /// This however has some drawbacks:
+  /// - Only packages with licenses are known
+  /// - No version information is available
+  /// - Flutter's native dependencies are also included.
+  FutureOr<Map<String, String>> _getPackages() async {
+    if (_packages.isEmpty) {
+      // This can take some time.
+      // Therefore we cache this after running
+      var packages = <String>{};
+      await LicenseRegistry.licenses.forEach(
+        (entry) => packages.addAll(
+          entry.packages.toList(),
```

Comment:
I probably should add a comment what's happening here:

The license registry has a list of licenses entries (MIT, Apache...).
Each license entry has a list of packages which licensed under this particular license.
Libraries can be dual licensed.

We don't care about those license issues, we just want each package name once.
Therefore I add each name to a set to make sure I only have it once.

Reply:
sounds good, is that the only way to get the packages list? because as. you said, if a package does not have a package, it will not appear there